src/waypoint.jsx

```diff
@@ -34,11 +34,24 @@ function getCurrentPosition(bounds) {
     return POSITIONS.invisible;
   }
 
+  // top is within the screen
   if (bounds.viewportTop <= bounds.waypointTop &&
       bounds.waypointTop <= bounds.viewportBottom) {
     return POSITIONS.inside;
   }
 
+  // bottom is within the screen
+  if (bounds.viewportTop <= bounds.waypointBottom &&
+      bounds.waypointBottom <= bounds.viewportBottom) {
+    return POSITIONS.inside;
+  }
+
+  // top is above the screen and bottom is below the screen
+  if (bounds.waypointTop <= bounds.viewportTop &&
+      bounds.viewportBottom <= bounds.waypointBottom) {
+    return POSITIONS.inside;
+  }
+
   if (bounds.viewportBottom < bounds.waypointTop) {
     return POSITIONS.below;
   }
```

Comment:
It may make sense to make the "outside" conditionals first. I suspect on a page with lots of waypoints, most of them will not be "inside," so this would allow us to skip over the other conditions. This may just be a micro-optimization with no real performance gain, though. What do y'all think?